src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -865,31 +789,29 @@ protected void Accepting()
             Receive<Send>(send =>
             {
                 var recipientAddress = send.Recipient.Path.Address;
-                IActorRef CreateAndRegisterWritingEndpoint(int? refuseUid) => _endpoints.RegisterWritableEndpoint(recipientAddress, CreateEndpoint(recipientAddress, send.Recipient.LocalAddressToUse, _transportMapping[send.Recipient.LocalAddressToUse], _settings, writing: true, handleOption: null, refuseUid: refuseUid), uid: null, refuseUid: refuseUid);
+                IActorRef CreateAndRegisterWritingEndpoint() => _endpoints.RegisterWritableEndpoint(recipientAddress, 
+                    CreateEndpoint(recipientAddress, send.Recipient.LocalAddressToUse, _transportMapping[send.Recipient.LocalAddressToUse], 
+                        _settings, writing: true, handleOption: null), uid: null);
 
                 // pattern match won't throw a NullReferenceException if one is returned by WritableEndpointWithPolicyFor
-                _endpoints.WritableEndpointWithPolicyFor(recipientAddress).Match()
-                    .With<Pass>(
-                        pass =>
-                        {
-                            pass.Endpoint.Tell(send);
-                        })
-                    .With<Gated>(gated =>
-                    {
-                        if (gated.TimeOfRelease.IsOverdue) CreateAndRegisterWritingEndpoint(gated.RefuseUid).Tell(send);
+                switch (_endpoints.WritableEndpointWithPolicyFor(recipientAddress))
```

Comment:
Rewrote policy handlers as `switch` statements - but otherwise functionality is the same / updated to match JVM.